framework/test/unit/controller/bus.spec.ts

```diff
@@ -46,34 +48,98 @@ describe('Bus', () => {
 		type: 'inMemory',
 		channel: channelMock,
 	};
-	const logger: any = {
+
+	const loggerMock: any = {
 		info: jest.fn(),
 		error: jest.fn(),
+		debug: jest.fn(),
 	};
 
 	let bus: Bus;
 
 	beforeEach(() => {
-		bus = new Bus(logger, config);
+		bus = new Bus(loggerMock, config);
 	});
 
-	afterEach(() => {
-		bus['_wsServer'].stop();
+	afterEach(async () => {
+		if (bus) {
+			await bus.cleanup();
+		}
 	});
 
 	describe('#constructor', () => {
 		it('should create the Bus instance with given arguments.', () => {
 			// Assert
 			expect(bus['actions']).toEqual({});
 			expect(bus['events']).toEqual({});
+			expect(bus['_ipcServer']).toBeInstanceOf(IPCServer);
 			expect(bus['_wsServer']).toBeInstanceOf(WSServer);
 		});
 	});
 
 	describe('#setup', () => {
+		beforeEach(() => {
+			jest.spyOn(IPCServer.prototype, 'start');
+			jest.spyOn(WSServer.prototype, 'start');
+		});
+
 		it('should resolve with true.', async () => {
 			return expect(bus.setup()).resolves.toBe(true);
 		});
+
+		// TODO: Should be tested in integration tests as the mock is complex to handle here
+		// eslint-disable-next-line jest/no-disabled-tests
+		it.skip('should setup ipc server if ipc is enabled', async () => {
+			// Arrange
+			const updatedConfig = { ...config };
+			updatedConfig.ipc.enabled = true;
+			bus = new Bus(loggerMock, updatedConfig);
+
+			// Act
+			await bus.setup();
+
+			// Assert
+			return expect(IPCServer.prototype.start).toHaveBeenCalledTimes(1);
```

Comment:
```suggestion
                         expect(WSServer.prototype.start).toHaveBeenCalledTimes(0);
			return expect(IPCServer.prototype.start).toHaveBeenCalledTimes(1);
```
just to make sure WSServer was never called? and same could be done for below 4 tests

Reply:
Both are atomic expectations so I prefer to test in separate blocks.

Reply:
🤔 i think testing if only ipc to start and ws to not start is still atomic? but i get your point about having separate test to check them